infra/terraform/README.md

```diff
@@ -0,0 +1,249 @@
+      - Create IAM users who will
+
+   use Terraform.
+      - Grant these users only the permission to assume the previously created role. Example policy for the user:
+      ```json
+      {
+         "Version": "2012-10-17",
+         "Statement": [
+            {
+            "Effect": "Allow",
+            "Action": "sts:AssumeRole",
+            "Resource": "arn:aws:iam::123456789012:role/terraform-role"
+            }
+         ]
+      }
+      ```
+
+   4. **Configure AWS CLI for Role Assumption:**
+      - Update the AWS CLI configuration to use the assumed role:
+      ```bash
+      aws configure set role_arn arn:aws:iam::123456789012:role/terraform-role --profile terraform-user
+      aws configure set source_profile default --profile terraform-user
+      ```
+
+   5. **Configure Terraform to Use the IAM User Profile:**
+      - Update your Terraform provider configuration in `provider.tf`:
+      ```hcl
+      provider "aws" {
+         profile = "terraform-user"
+         region  = "us-west-2"
+      }
+      ```
+
+   6. **Run Terraform Commands:**
+      - When you run Terraform commands, the AWS CLI will assume the specified role and use the permissions defined for that role.
+
+   ### Additional Resources
+
+   - [AWS IAM Best Practices](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html)
+   - [Terraform AWS Provider Security Best Practices](https://docs.aws.amazon.com/prescriptive-guidance/latest/terraform-aws-provider-best-practices/security.html)
+   - [Assuming IAM Roles with Terraform](https://developer.hashicorp.com/terraform/tutorials/aws/aws-assumerole)
+
+5. **Define Variables**
+
+   ```bash
+   variables.tf
+   ```
+   This file holds the variables of the Terraform configuration that can be modified as required by the user.
+
+6. **Run the Script**
+
+    ```bash
+    bash apply-script.sh
+    ```
+
+    Run this command to initiate, format, validate, and plan Terraform scripts.
+
+7. **Apply Infrastructure**
+
+    ```
+    terraform apply --auto-approve
+    ```
+    Run this command to apply the Terraform scripts to create infrastructure on your AWS account.
+
+8. **Logs Location**    
+    You can connect to the machine using **Session Manager** to check the logs at this location using this command:
+
+    ```
+    tail -f /var/log/nat_instance_setup.log
+    ```
+    
+    **Please wait for at least 15 minutes for Terraform to apply scripts and 30 minutes more for Docker containers to build and run.**
+
+    > Note: If you have not configured SMTP for your OWTF application, use the logs of the OWTF Docker container to get the verification link during login. Make sure to replace the appropriate IP address in that link with the **ALB DNS Name**. 
+    
+    Use this command to get the logs of the Docker container inside the EC2 instance:
+
+    ```
+    docker logs <Container_ID/Container_Name>
+    ```
+
+9. **Destroy Infrastructure**
+
+    ```
+    terraform destroy --auto-approve
```

Comment:
can you remove `--auto-approve` from the tf apply and destroy commands? Ideally Terraform warns if some unintended effect takes place and auto approving doesn't help.